brave-spark-framework-filter/src/test/java/com/github/kristofa/brave/spark/ITBraveSparkTest.java

```diff
@@ -0,0 +1,67 @@
+package com.github.kristofa.brave.spark;
+
+import com.github.kristofa.brave.Brave;
+import com.github.kristofa.brave.LocalTracer;
+import com.github.kristofa.brave.http.ITHttpServer;
+import com.github.kristofa.brave.http.SpanNameProvider;
+import org.junit.After;
+import org.junit.AssumptionViolatedException;
+import org.junit.Test;
+import spark.*;
+
+/**
+ * Created by 00013708 on 2017/3/17.
+ */
+public class ITBraveSparkTest extends ITHttpServer {
+    /*after filter executed before NotFoundRoute,there is no way to add 404 code after serverspan
+    has been setServerSend,may be throw a not found exceeption and process it in ExceptionHandler*/
+    @Override
+    @Test
+    public void addsStatusCodeWhenNotOk() throws Exception {
+        throw new AssumptionViolatedException("TODO: fix error reporting");
+    }
+
+    @Override
+    protected void init(Brave brave, SpanNameProvider spanNameProvider) throws Exception {
+        stop();
+        Spark.init();

```

Comment:
1.OK,I will have a try using test async.
2.Spark.awaitInitialization() is a method blocks wait spark initialization finished.I had used it ,but some problems happen.There should be a method for Spark.waitStop().
https://github.com/perwendel/spark/pull/730